src/vehicle.cpp

```diff
@@ -3433,36 +3433,17 @@ int vehicle::basic_consumption( const itype_id &ftype ) const
     return fcon;
 }
 
-int vehicle::consumption_per_hour( const itype_id &ftype, int fuel_rate_w ) const
+long long vehicle::consumption_per_hour( const itype_id &ftype, int fuel_rate_w ) const
```

Comment:
[Use `int64_t`](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CODE_STYLE.md#code-guidelines), but is `fuel_rate_w` really so close to `INT_MAX` that multiplying it by 3.6 (and probably even less) that it's going to overflow?

Reply:
I wasn't sure initially but I think thats due to how I had the constants. The cases causing overflow were for the larger engines at full capacity. Anything producing more that about 600 kW power would cause that. If I consolidate the constants I can probably walk this back.

Reply:
It's also valid to use a larger type for the calculations, then return a smaller type (though CI/compilers may be a bit annoying about casting to smaller types).

Reply:
32 bit integers should be able to support ~1/2 a DeLorean (580 MW) before overflowing. Thats commercial nuclear reactor range of power output. I think that should be plenty.